File: app/src/main/java/com/damhoe/scoresheetskat/score/domain/SkatSuit.java
package com.damhoe.scoresheetskat.score.domain;

public enum SkatSuit {
 GRAND,
 NULL,
 CLUBS,
 SPADES,
 HEARTS,
 DIAMONDS,
 INVALID;

 public int asInteger() {
  return ordinal();
 }

 public static SkatSuit fromInteger(int value) {
  for (SkatSuit suit: SkatSuit.values()) {
   if (suit.ordinal() == value) {
    return suit;
   }
  }
  return SkatSuit.CLUBS; // default
 }
}

File: app/src/main/java/com/damhoe/skatscores/score/domain/SkatSuit.kt
package com.damhoe.skatscores.score.domain

enum class SkatSuit {
    GRAND, NULL, CLUBS, SPADES, HEARTS, DIAMONDS, INVALID;

    fun asInteger(): Int = ordinal

    companion object {
        fun fromInteger(value: Int): SkatSuit = values().find { it.ordinal == value } ?: CLUBS
    }
}